packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -68,62 +101,76 @@ class _BannerMessageContainerState extends State<BannerMessageContainer>
     final newController = Controllers.of(context)?.bannerMessages;
     if (newController == controller) return;
     controller = newController;
+  }
+
+  void push(BannerMessage message) {
+    if (controller.isMessageDismissed(message) ||
+        controller.isMessageVisible(message)) return;
+    // We push the banner message in a post frame callback because otherwise,we'd be
+    // trying to call setState while the parent widget `BannerMessages` is in the middle
+    // of `build`.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (!mounted) return;
+      setState(() {
+        controller.addMessage(message);
+      });
+    });
+  }
 
-    autoDispose(controller.onRefreshMessages.listen((_) => setState(() {})));
+  void remove(BannerMessage message, {bool dismiss = false}) {
+    // We push the banner message in a post frame callback because otherwise,we'd be
+    // trying to call setState while the parent widget `BannerMessages` is in the middle
+    // of `build`.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (!mounted) return;
+      setState(() {
+        controller.removeMessage(message, dismiss: dismiss);
```

Comment:
as above. having a set state around a call to a controller is a sign this widget isn't listening for changes to the controller. 
For example, this is broken if you happen to have two banner message widgets listening on the same controller.